synapse/config/server.py

```diff
@@ -55,6 +55,15 @@ def read_config(self, config):
             "block_non_admin_invites", False,
         )
 
+        federation_domain_whitelist = config.get(
```

Comment:
It feels very odd that you can whitelist everything by having an empty whitelist. I wonder if it would be better to make an empty whitelist actually mean that nothing is allowed, and have the default be `*`

Reply:
good point; fixed.